### PR TITLE
Guide: Modeling complex workflows

### PR DESCRIPTION
**Pull request by bmarkons**
This the guide I wrote on how to model complex CI/CD flows with Semaphore pipelines.

Waiting for feedback. 🙇

### CODE REVIEW

File: next-gen-pipeline-guide.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+`dependencies` property since there isn't any block which code checks should
+depend on.
+
+```
+  - name: "Code quality checks"
+    dependencies: []
+
+```
+
+*Note*: In case any of your blocks can start immediately without depending on any other
```
Comment (darkofabijan) [inline next-gen-pipeline-guide.md]: rm -rf "Note:"

**Comment by bmarkons** [comment]
I will use the example above as representative one and modify the guide in order to address other points as well.

Thank you for suggestions @markoa, your feedback is hugely appreciated.

**Comment by bmarkons** [comment]
I have made a number of changes but I would like to outline the following:

- Titled the guide with: Modeling complex workflows
- Added real world workflow scenario just below the title.
- Added a very simple therefore straightforward example for people to get an idea how to define dependencies.
- Also included Fan-in/Fan-out, Monorepo and Cross-platform examples as common use cases.

@darkofabijan created a new repo [here](https://github.com/semaphoreci-demos/semaphore-demo-workflows) where we can store fully functional YAMLs for examples we used in this guide.

What do you think @markoa?

**Comment by markoa** [comment]
@bmarkons Great work! Reading the guide is a very nice experience now.

I've made some style edits and pushed a new version.

**Comment by bmarkons** [comment]
Thanks, @markoa!

In the meanwhile, the article has been created and the document is filled with the links to the full yaml specifications which are used in examples.

Can you merge this? Once you do, we need to publish the article.

**Comment by markoa** [comment]
👏 

Published: https://docs.semaphoreci.com/article/144-modeling-complex-workflows

Please fix the image URLs. Don't wait for my approval to merge.

**Pull request merged**